server/api/messaging.py

```diff
@@ -0,0 +1,207 @@
+    # Try to get all received messages
+    # if no instances are found, catch 
+    # the exception and tell the user 
+    # there are no messages
+    try:
+        messages = Message.get_instance(multiple=True, **{'receiver': user})
+    except NoResultFound:    
+        response_dict['status'] = 204
+        response_dict['message'] = "You haven't received any messages yet"
+        return jsonify(response_dict), 204
+
+    for message in messages:
+        message_dict = message.to_dict(exclude=['receiver', 'sender', 'created_at'])
+        message_dict['sender'] = message.sender.profile.name
+        message_dict['sender_image_url'] = message.sender.profile.profile_image
+        for field, value in message.get_formatted_info.items():
+            message_dict[field] = value
+        response_dict['messages'].append(message_dict)
+
+    return jsonify(response_dict), 200
+
+@messaging_views.route('/get_sent_messages', methods=["GET"])
+@authenticate
+def sent_messages(**kwargs):
+    response_dict = {'messages': []}
+    user = kwargs['user']
+    try:
+        messages = Message.get_instance(multiple=True, **{'sender': user})
+    except NoResultFound:    
+        response_dict['status'] = 204
+        response_dict['message'] = "You haven't sent any messages yet"
+        return jsonify(response_dict), 204
+
+    for message in messages:
+        message_dict = message.to_dict(excludes=['sender', 'receiver', 'created_at'])
+        for field, value in message.get_formatted_info.items():
+            message_dict[field] = value
+        message['receiver'] = message.receiver.profile.name
+        message['receiver_image'] = message.receiver.profile.profile_image
+        response_dict['messages'].append(message_dict)
+    return jsonify(response_dict), 200
+
+
+@messaging_views.route('/get_conversation_messages', methods=["GET"])
+@authenticate
+def conversation_messages(**kwargs):
+    """
+    Get all messages between the logged in user
+    and the other user then order them by the 
+    created_at field to ensure a proper descending
+    order so the newest messages are at the bottom
+    """
+    response_dict = {'messages': []}
+    user = kwargs['user']
+    other_user_id = request.get_json().get('other_user_id')
+    try:
+        other_user = User.get_instance(**{'id': other_user_id})
+    except NoResultFound:
+        response_dict['status'] = 404
+        response_dict['message'] = "User with id %s does not exist" % (other_user_id)
+        return jsonify(response_dict), 404
+    # Get the conversation between the two users.
+    conversation = Conversation.get_conversation(user, other_user)
+    # Messages in the conversation are already ordered by descending created_at (date/time)
+    for message in conversation.messages.all():
+        if message.sender != user:
+            profile_image = message.sender.profile.profile_image
+        else:
+            profile_image = message.receiver.profile.profile_image
```

Comment:
the current user profile image should already be available on the frontend. The other user profile image makes sense to be included when returning a conversation itself instead of with every message, we only need that information once.
Not to mention there could be lots of messages in a conversation, probably best not to loop over them all

Reply:
Very good point, I don't even know what I was thinking. I could probably just get the other users image from the conversation anyway. 
About not looping through all the messages, we would still need the information on each message to be serialized right? Or are you saying to just remove this check from the loop?
Or should I attempt pagination and load more when the user scrolls past what's already loaded?